src/components/dialogs/parameters/sensi/sensitivity-table.js

```diff
@@ -0,0 +1,121 @@
+/**
+ * Copyright (c) 2023, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+import {
+    Table,
+    TableBody,
+    TableCell,
+    TableContainer,
+    TableHead,
+    TableRow,
+} from '@mui/material';
+import Tooltip from '@mui/material/Tooltip';
+import IconButton from '@mui/material/IconButton';
+import AddCircleIcon from '@mui/icons-material/AddCircle';
+import React from 'react';
+import { FormattedMessage, useIntl } from 'react-intl';
+import { MAX_ROWS_NUMBER } from '../../../utils/dnd-table/dnd-table';
+import { useFormContext } from 'react-hook-form';
+import { Box } from '@mui/system';
+import TableRowComponent from './table-row';
+
+const SensitivityTable = ({
+    arrayFormName,
+    useFieldArrayOutput,
+    columnsDefinition,
+    tableHeight,
+    createRows,
+}) => {
+    const intl = useIntl();
+    const { getValues, setError, clearErrors } = useFormContext();
+
+    const { fields: currentRows, append, remove } = useFieldArrayOutput;
+
+    function handleAddRowsButton() {
+        let numberOfRows = 1;
+        if (currentRows.length + numberOfRows > MAX_ROWS_NUMBER) {
```

Comment:
constant, be ca removed
```suggestion
        if (currentRows.length + 1 > MAX_ROWS_NUMBER) {
```

Reply:
In this code, numberOfRows remains a variable, allowing us to generate rows dynamically based on its value. The condition checks if adding numberOfRows to the current number of rows in currentRows exceeds the MAX_ROWS_NUMBER. If it does not exceed the maximum allowed rows, it proceeds to generate rows using the createRows function and stores them in the rowsToAdd variable. we can continue with the code logic using the rowsToAdd variable. This approach maintains the dynamic behavior of generating rows while also considering the maximum allowed rows.